independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/ServerEndpointIndexer.java

```diff
@@ -300,9 +300,16 @@ protected boolean handleBeanParam(ClassInfo actualEndpointInfo, Type paramType,
                 actualEndpointInfo,
                 existingConverters, additionalReaders, injectableBeans, hasRuntimeConverters);
         if ((injectableBean.getFieldExtractorsCount() == 0) && !injectableBean.isInjectionRequired()) {
-            throw new DeploymentException(String.format("No annotations found on fields at '%s'. "
-                    + "Annotations like `@QueryParam` should be used in fields, not in methods.",
-                    beanParamClassInfo.name()));
+            long declaredMethodsCount = beanParamClassInfo.methods().stream()
```

Comment:
You're counting methods but the message is about fields. I'm not sure I understand the logic here?

Reply:
Exactly. The current error message is misleading when there are no methods **and** fields so this tries to make things less misleading